Exercise 1 BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,216 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.*;
+import javax.baja.sys.*;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "fileName",
+        type = "String",
+        defaultValue = "",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "path",
+        type = "String",
+        defaultValue =  "",
+        flags = Flags.SUMMARY
+)
+@NiagaraProperty(
+        name = "size",
+        type = "int",
+        defaultValue = "0",
+        facets = {
+                @Facet(name = "BFacets.MIN", value = "0")
+        }
+)
+@NiagaraProperty(
+        name = "createdDate",
+        type = "baja:AbsTime",
+        defaultValue = "BAbsTime.NULL"
+)
+@NiagaraProperty(
+        name = "modifiedDate",
+        type = "baja:AbsTime",
+        defaultValue = "BAbsTime.NULL"
+)
+@NiagaraAction(
+        name = "touch"
+)
+@NiagaraAction(
+        name = "print"
+)
+
+
+public class BFile extends BComponent {
+
+  public void doPrint(){
+    System.out.println("Properties for " + getFileName() + "\nPath: " + getPath() + "\nSize: " + getSize() +" Bytes\nDate Created: "
+            + getCreatedDate() + "\nDate Modified: " + getModifiedDate());
+  }
+
+  public void doTouch(){
+    setModifiedDate(BAbsTime.now());
+    System.out.println("Touching now");

```

Comment:
This isn't a big deal while testing the code, but make sure not to leave these kinds of statements in production code. A user wouldn't expect to see "Touching now" in the station output whenever they invoke the touch Action.